ledger/ledger-common/src/main/java/net/corda/v5/ledger/common/NotaryLookup.java

```diff
@@ -0,0 +1,47 @@
+package net.corda.v5.ledger.common;
+
+import net.corda.v5.base.annotations.DoNotImplement;
+import net.corda.v5.base.annotations.Suspendable;
+import net.corda.v5.base.types.MemberX500Name;
+import net.corda.v5.membership.NotaryInfo;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+
+/**
+ * Defines a mechanism to allow flows to retrieve the NotaryInfo in the network.
+ * The platform will provide an instance of NotaryLookup to flows via property injection.
+ */
+@DoNotImplement
+public interface NotaryLookup {
+
+    /**
+     * Gets a {@link Collection} of {@link NotaryInfo} services available on the network.
+     *
+     * @return Returns a {@link Collection} of {@link NotaryInfo} services available on the network.
+     */
+    @NotNull
+    @Suspendable
+    // TODO : Consider List<NotaryInfo>
+    Collection<NotaryInfo> getNotaryServices();
```

Comment:
Re-raising of https://github.com/corda/corda-api/pull/864#discussion_r1120045788 to ensure this is not lost due to the new PR being raised.

Can we address the philosophy of why this should be a `List`, or left as a `Collection`? `Collection` could presumably be specialised in future as a non-breaking API change if there was really a need for it (since everything ultimately inherits from `Collection`

Reply:
I would say if we do leave it as a collection, we should probably remove the TODO - as it seems unlikely we will revisit in any specific timeframe, it would only change if there became a concrete need for it down the line.